cleanWeather-vue/4.md

```diff
@@ -0,0 +1,106 @@
+## Lesson 4. Querying an API to get a Weather Forecast. 
+
+In this chapter we will deal with the second section of our application, which in a Vue.js app is often contained in the `<script>` block of a Single File Component.
+
+Right under the `<script>` tag, add the following three lines, which make the Geolocation package available for your app. Similarly, your app will need to set the accuracy of the data, an element that is available in the `ui/enums` package. Finally, you're going to need to use the `http` module to query the OpenWeatherMap API for data.
+
+```
+const Geolocation = require("nativescript-geolocation");
+const Accuracy = require("ui/enums"); 
+const http = require("http");
+```
+
+Now you can start working on the `data()` method in the `<script>` block. You need to set some default variables so that your screen will be properly populated after the API is queried. Replace the current `data()` method with the following:
+
+```
+data () {
+        return {
+			 image: '', 
+             city: '', 
+             windSpeed: '', 
+             summary: '', 
+             visibility: '', 
+             humidity: '', 
+             currentTemperature: ''
+        };
+    },
+```
+
+> **TIP**:
+> * If your pasted code looks unformatted in the Playground, try `ctrl-Z` to reformat it.
+
+Next, we will add a lifecycle hook, the method `created()`. Under the final comma closing the `data()` method, add a `created()` method that will set some variables needed in the app:
+
+```
+created() { 
+    //this.getMyWeather() 
+    var currentDate = new Date() 
+    var day = currentDate.getDay()
+	var weekdays = new Array(7); 
+    weekdays[0] = "Sunday"; 
+    weekdays[1] = "Monday"; 
+    weekdays[2] = "Tuesday"; 
+    weekdays[3] = "Wednesday";
+	weekdays[4] = "Thursday"; 
+    weekdays[5] = "Friday"; 
+    weekdays[6] = "Saturday"; 
+    var dayName = weekdays[day]; 
+    var currentHours = currentDate.getHours(); 
+    var timeOfDay = (currentHours >= 12 ) ? "Afternoon" : "Morning"; 
+	this.day = dayName;
+    this.time = timeOfDay; 
+},
+```
+
+When your app refreshes on device, you'll find two of its elements are already populated, the dayName and timeOfDay. It doesn't look great, but you'll fix that soon.
+
+Before you continue, you need to register with [OpenWeatherMap](http://www.openweathermap.org). Sign up for a free account and create an API key. Your API key may take up to ten minutes to be activated, so be a bit patient. When it is activated, make a note of it as you'll need it below.
```

Comment:
This should probably be in an action block too to make it clear the user is supposed to do something here.